src/popup/router/components/TransactionOverview.vue

```diff
@@ -0,0 +1,272 @@
+<template>
+  <div class="transaction-overview">
+    <div class="title">
+      {{ transaction.title }}
+      <span class="description">({{ transaction.description }})</span>
+    </div>
+
+    <div class="parties">
+      <Avatar v-bind="transaction.sender" />
+      <div class="mid">
+        <TriangleRight />
+        <div class="line" />
+      </div>
+      <div v-if="transaction.aens" class="avatar">
+        <AensIcon class="icon" />
+      </div>
+      <div v-else-if="transaction.contractCreate" class="avatar">
+        <ActionIcon class="icon" />
+      </div>
+      <Avatar v-else-if="transaction.recipient" v-bind="transaction.recipient" />
+      <div v-else class="avatar">
+        <ActionIcon class="icon" />
+      </div>
+    </div>
+
+    <div class="details">
+      <div class="sender">
+        <a
+          :href="connect ? app.host : getExplorerPath(transaction.sender.address)"
+          target="_blank"
+          class="name"
+        >
+          <TruncateMid :str="transaction.sender.name || transaction.sender.label" />
+        </a>
+        <CopyAddress :value="transaction.sender.address" />
+      </div>
+      <div class="recipient">
+        <span
+          v-if="transaction.aens || transaction.contractCreate"
+          :class="{ aens: transaction.aens }"
+          class="name"
+        >
+          {{ transaction.recipient.label }}
+        </span>
+        <a
+          v-else
+          :href="getExplorerPath(transaction.recipient.address)"
+          target="_blank"
+          class="name"
+        >
+          <TruncateMid :str="transaction.recipient.name || transaction.recipient.label" />
+        </a>
+        <CopyAddress v-if="transaction.recipient.address" :value="transaction.recipient.address" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex';
+import { TX_TYPE, OBJECT_ID_TX_TYPE } from '@aeternity/aepp-sdk/es/tx/builder/schema';
+import { TX_TYPE_MDW } from '../../utils/constants';
+import TriangleRight from '../../../icons/triangle-right.svg?vue-component';
+import ActionIcon from '../../../icons/action.svg?vue-component';
+import AensIcon from '../../../icons/aens.svg?vue-component';
+import Avatar from './Avatar';
+import TruncateMid from './TruncateMid';
+import CopyAddress from './CopyAddress';
+
+export default {
+  components: {
+    Avatar,
+    TruncateMid,
+    CopyAddress,
+    TriangleRight,
+    AensIcon,
+    ActionIcon,
+  },
+  props: {
+    tx: { type: Object, default: null },
+    connect: { type: Boolean, default: false },
+    app: { type: Object, default: null },
```

Comment:
Splitting this component into Overview, TransactionOverview, ConnectOverview (?), would allow to make it more readable and to check props stricter

Reply:
I extracted Overview from TransactionOverview. Connect modal can use Overview with little adaptation.